### DISCUSSION TITLE
Milestone planning for Akka.Management version 1.0

### DISCUSSION THREAD

**Discussion by Arkatufus** [akkadotnet/Akka.Management#417] [Ideas]
# Milestones for 1.0

Some things are still missing from this module as a whole:
* [x] Amazon AWS ECS discovery
  Will need to  port this manually, I think I hit a brick wall when I tried to port this but couldn't remember why.

~~Consul discovery,~~
~~still need to move this from @Horusiath separate repo, should be an easy port.~~

~~Marathon discovery~~
~~Need to learn about Marathon and figure out if a port is feasible.~~

* [ ] Azure Blob Storage lease
  Will have to write this from scratch, it shouldn't be that hard.

* [x] We might consider replacing Kestrel and ASP,NET with something a lot more light weight.
List of available light weight pure dotnet web server, ommiting confirmed dead ones:
  * [EmbedIO](https://github.com/unosquare/embedio): this seemed to be still under active development, lots of reported active users
  * Separate Kestrel from ASP.NET: this seems like a very arduous thing to do since Kestrel is tightly coupled to ASP.NET, but it seems like this is doable as demonstrated [here](https://github.com/PeteX/StandaloneKestrel)
  * [uHttpSharp](https://github.com/bonesoul/uhttpsharp): have not been updated in 3 years
  * [Ceen.HTTPd](https://github.com/kenkendk/ceenhttpd): maybe still under active development, last update was 6 months ago
  * [NetCoreServer](https://github.com/chronoxor/NetCoreServer): still under active development, decent active users
  * [Watson Webserver](https://github.com/jchristn/WatsonWebserver): still under active development.
  * [GenHTTP](https://github.com/Kaliumhexacyanoferrat/GenHTTP): still under active development.

  Will need to weed out the list and try to pick one that will work in all production environments. Selection criteria should be:
    * Supports NetStandard 2.0
    * Small memory footprint
    * SSL capable for future proofing
    * Platform agnostic
    * Still in active development, we would not want to have to maintain them in the future if the maintainer decided to stop supporting the project

**Comment by Aaronontheweb** [comment]
Drop Consul and Marathon discovery for 1.0. Those can be added later and I'm pretty sure Marathon is toast anyway.

Drop Akka.HTTP - keep the shim, turn proper Akka.HTTP into its own project in the future. Didn't test well in our survey

**Comment by Arkatufus** [comment]
Replaced the internal server implementation for Akka.Http with Ceen, this is to avoid any collision with Microsoft IIS when an application is being deployed "in-process"
